src/shared/Core/CredentialStore.cs

```diff
@@ -107,7 +107,7 @@ private void EnsureBackingStore()
                     AppendAvailableStoreList(sb);
                     sb.AppendLine();
                     sb.AppendLine($"See {Constants.HelpUrls.GcmCredentialStores} for more information.");
-                    throw new Exception(sb.ToString());
+                    throw new Trace2Exception(_context.Trace2, sb.ToString());
```

Comment:
I'm wondering if we really want a multiline message with hint text in the trace log ??   Especially, the hint text.

Perhaps we should call `WriteError()` after the initial message string is inserted into the buffer (between lines 100 and 101) and before the hint text is appended.  Then just throw using the original Exception() class.